components/dashboard/src/service/service.tsx

```diff
@@ -56,7 +58,67 @@ export function getGitpodService(): GitpodService {
         const service = _gp.gitpodService || (_gp.gitpodService = require("./service-mock").gitpodServiceMock);
         return service;
     }
-    const service = _gp.gitpodService || (_gp.gitpodService = createGitpodService());
+    let service = _gp.gitpodService;
+    if (!service) {
+        let user: any;
+        service = _gp.gitpodService = createGitpodService();
+        service.server = new Proxy(service.server, {
+            get(target, propKey) {
+                return async function (...args: any[]) {
+                    if (propKey === "getLoggedInUser") {
+                        user = await target[propKey](...args);
+                        return user;
+                    }
+                    if (propKey === "getWorkspace") {
+                        try {
+                            return await target[propKey](...args);
+                        } finally {
+                            const grpcType = "unary";
+                            // emulates frequent unary calls to public API
+                            const isTest = await getExperimentsClient().getValueAsync(
+                                "public_api_dummy_reliability_test",
+                                false,
+                                { user },
```

Comment:
Can we add this (same as here):
```suggestion
                                { user, gitpodHost: window.location.host },
```
I thought we would inject this in the proxy route, but it seems we do it in the query.